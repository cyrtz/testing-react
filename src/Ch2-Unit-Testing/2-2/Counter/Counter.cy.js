import React from 'react';
import Counter from './Counter';

describe('Counter', () => {
    it('should get 1 when click increment button', () => {
        cy.mount(<Counter />);
        cy.get('[data-test-id="increment-button"]').click();
        cy.get('[data-test-id="counter-value"]').should('have.text', '1');
    });
});