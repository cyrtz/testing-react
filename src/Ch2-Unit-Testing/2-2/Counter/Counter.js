import { useState } from "react";
import React from "react";

const Counter = () => {
    const [count, setCount] = useState(0);
    const increment = () => setCount(count + 1);
    const decrement = () => setCount(count - 1);

    return (
        <>
            <div data-test-id="counter-value">{count}</div>
            <button data-test-id="increment-button" onClick={increment}>+</button>
            <button data-test-id="decrement-button" onClick={decrement}>-</button>
        </>
    )
}

export default Counter;