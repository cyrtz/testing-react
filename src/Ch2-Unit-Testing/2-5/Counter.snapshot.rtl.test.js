import React from "react";
import Counter from './Counter-enhanced';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Counter Component', () => {
    xit('should match snapshot', () => {
        const { asFragment } = render(<Counter />);
        expect(asFragment()).toMatchSnapshot();
    });

    describe('full rendering', () => {
        let counter;
        const renderCounter = () => <Counter />;

        counter = render(renderCounter());

        it('should get 1 when click the increment button', () => {
            const { getByTestId } = counter;
            fireEvent.click(screen.getByTestId('increment-button'));
            expect(screen.getByTestId('counter-value')).toHaveTextContent('1');
        });
    });
});