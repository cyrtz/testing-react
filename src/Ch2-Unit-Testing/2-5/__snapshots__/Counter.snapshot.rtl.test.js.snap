// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component full rendering should get snapshop correctly 1`] = `
<div>
  <p
    data-test-id="counter-value"
  >
    0
  </p>
  <button
    data-test-id="decrement-button"
  >
    -
  </button>
  <button
    data-test-id="increment-button"
  >
    +
  </button>
</div>
`;

exports[`Counter Component should match snapshot 1`] = `
<DocumentFragment>
  <p
    data-test-id="counter-value"
  >
    0
  </p>
  <button
    data-test-id="decrement-button"
  >
    -
  </button>
  <button
    data-test-id="increment-button"
  >
    +
  </button>
</DocumentFragment>
`;
